### PR TITLE
FSE: Remove page title block when editing template parts

### PR DESCRIPTION
**Pull request by glendaviesnz**
#### Changes proposed in this Pull Request

* Remove the post title when editing a template part in FSE

#### Testing instructions

* Check out this branch and rebuild the FSE plugin for your local FSE test environment `npx lerna run dev --scope='@automattic/full-site-editing'`
* Edit a Header or Footer template part and check that the page title block does not display in the editor

**Before:**
before

**After**
after

Part of #34728

### CODE REVIEW

**Comment by Copons** [comment]
Is there a way to maybe sort this out with CSS only? 🤔 

---

Wouldn't this risk disorienting the user?
By removing the title, we wouldn't have anything telling us what we are editing (well, except the template part type taxonomy).
It could be ok now that we only have 1 header and 1 footer, but what about when we start introducing multiple template parts?

To be honest, I'd rather keep the title and, eventually, place it outside the canvas proposed by @shaunandrews in https://github.com/Automattic/wp-calypso/pull/34684

**Comment by shaunandrews** [comment]
> By removing the title, we wouldn't have anything telling us what we are editing

We'll have a notice here telling customers they're editing a header, which is rendered on every page of their site. Having the title block here just litters the preview of what they're editing, and we don't want them to be able to edit the title anyways.

> To be honest, I'd rather keep the title and, eventually, place it outside the canvas

This is a "nice to have," but I don't think we're anywhere close to having it. For now, the title in these template_parts is useless, confusing, and needs to go.

**Comment by glendaviesnz** [comment]
> Is there a way to maybe sort this out with CSS only?

Good question, will take another look at it and see if we can target some existing classes.

**Comment by glendaviesnz** [comment]
> Is there a way to maybe sort this out with CSS only?

@Copons - so I am sure that yesterday when I looked there were no unique selectors to identify that we were in a template part, but today when I looked there on the body is `.post-type-wp_template_part`! So I have updated this to a css only solution. @vindl - am I correct that `wp_template_part` is a custom post type currently unique to FSE?

File: apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/style.scss [hunk 1]
```diff
@@ -0,0 +1,5 @@
+.post-type-wp_template_part {
+	.editor-post-title {
+		display: none;
```
Comment (gwwar) [inline apps/full-site-editing/full-site-editing-plugin/full-site-editing/editor/style.scss]: 💯

**Comment by vindl** [comment]
> @vindl - am I correct that wp_template_part is a custom post type currently unique to FSE?

As far as I know it is, unless there is some other plugin in the wild that uses it - although I would find it unlikely that they use `wp` prefix and not their plugin name for instance.

**Pull request merged**